Amazon.SQS/src/Extensions.SQS/Collector/SqsQueueAsyncCollector.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Amazon;
+using Amazon.Runtime;
+using Amazon.SQS;
+using Azure.Functions.Extensions.SQS.Commons;
+using Microsoft.Azure.WebJobs;
+
+namespace Azure.Functions.Extensions.SQS.Collector
+{
+	public class SqsQueueAsyncCollector : IAsyncCollector<SqsQueueMessage>
+	{
+		private AmazonSQSClient AmazonSQSClient { get; }
+		private SqsQueueTriggerAttribute TriggerParameters { get; }
+		
+		public SqsQueueAsyncCollector(SqsQueueTriggerAttribute triggerParameters)
+		{
+			this.TriggerParameters = triggerParameters;
+			this.AmazonSQSClient = AmazonSQSClientFactory.Build(triggerParameters);
+
+		}
+		public async Task AddAsync(SqsQueueMessage item, CancellationToken cancellationToken = new CancellationToken())
+		{
+			await AmazonSQSClient.SendMessageAsync(item.QueueUrl ?? this.TriggerParameters.QueueUrl, item.Body, cancellationToken);
```

Comment:
All the parameters should be coming from one place IMO. I'd say just use the TriggerParameters for Url (after all client has to specify AWS credentials there). QueueMessage would include just the body and properties of the message.

Reply:
I see why you did it this way. This would allow clients to push messages to a different queue than the one they used in the trigger. I can see the benefit of that.